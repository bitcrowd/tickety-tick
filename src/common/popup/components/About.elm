module About exposing (..)

import Html exposing (..)
import Html.Attributes exposing (..)
import Html.Events exposing (..)

import ExternalLink

import Messages exposing (Msg(Navigate))
import Routes exposing (Route(IndexRoute))


view : Html Msg
view =
  div []
    [ div [ class "navbar navbar-light navbar-fixed-top navbar-tt" ]
      [ div [ class "nav navbar-nav" ]
        [ a [ class "nav-link", href "#index", onClick (Navigate IndexRoute) ] [ text "< back" ]
        ]
      ]
    , div [ class "content" ]
      [ div [ class "container" ]
        [ h1 [ class "h3 m-t-1 m-b-1" ]
          [ img [ src "../icons/icon-32.png", class "logo-sm", attribute "role" "presentation" ] []
          , text " tickety-tick"
          ]
        , h6 [] [ text "Usage:" ]
        , ol [ class "m-t-1 p-l-2" ]
            [ li [] [ text "Open one ore more tickets in your favourite ticket tracking system, then click the extension icon." ]
            , li [] [ text "You can then select if you want to create a commit message or a branch name based on these tickets." ]
            ]
        , p []
            [ text "This extension is open-source software by the fellows at "
            , extlink "http://bitcrowd.net" "bitcrowd"
            , text "."
            ]
        , p []
            [ text "The source code is available on "
            , extlink "https://github.com/bitcrowd/tickety-tick" "GitHub"
            , text "."
            ]
        , p [ class "small" ]
            [ text "Logo by "
            , extlink "http://thenounproject.com/term/ticket/92194/" "RamÃ³n G."
            , text " under CC-BY 3.0"
            , br [] []
            , text "Other icons from "
            , extlink "https://octicons.github.com/" "GitHub Octicons"
            ]
          ]
        ]
      ]

extlink = ExternalLink.view
