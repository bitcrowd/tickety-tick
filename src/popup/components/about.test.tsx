import { shallow } from "enzyme";
import React from "react";

import type { Props } from "./about";
import About from "./about";

describe("about", () => {
  function render(props: Props) {
    return shallow(<About {...props} />);
  }

  it("renders", () => {
    expect(render({})).toMatchSnapshot();
  });
});
