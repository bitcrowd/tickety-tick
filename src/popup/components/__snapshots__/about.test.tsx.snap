// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about renders 1`] = `
<DocumentFragment>
  <div
    class="navbar navbar-expand navbar-light bg-light border-bottom"
  >
    <div
      class="container-fluid"
    >
      <ul
        class="navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link ps-0"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="octicon-nav"
              fill="currentColor"
              focusable="false"
              height="14"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 12 12"
              width="14"
            >
              <path
                d="M3.587 6.025c0 .2.1.4.2.5l3.3 3.3c.3.3.8.3 1.1 0 .3-.3.3-.8 0-1.1l-2.7-2.7 2.7-2.7c.3-.3.3-.8 0-1.1-.3-.3-.8-.3-1.1 0l-3.2 3.2c-.2.2-.3.4-.3.6Z"
              />
            </svg>
            Back
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="container-fluid py-3"
  >
    <h1
      class="d-flex align-items-center h3 mb-3"
    >
      <svg
        class="logo me-1"
      >
        icon.svg
      </svg>
       Tickety-Tick
    </h1>
    <h6>
      Usage:
    </h6>
    <ol
      class="ps-3"
    >
      <li>
        Open a ticket in your favourite issue tracking system.
      </li>
      <li>
        Click the extension icon.
      </li>
      <li>
        Use the buttons to create a commit message, branch name or CLI command.
      </li>
    </ol>
    <p>
      This extension is open-source software by the fellows at 
      <a
        href="http://bitcrowd.net"
        rel="noopener noreferrer"
        target="_blank"
      >
        bitcrowd
      </a>
      .
    </p>
    <p>
      The source code is available on 
      <a
        href="https://github.com/bitcrowd/tickety-tick"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
      .
    </p>
    <p
      class="small"
    >
      <a
        href="https://github.com/bitcrowd/tickety-tick/tree/test-commit-hash"
        rel="noopener noreferrer"
        target="_blank"
        title="Browse source code for this version"
      >
        test-commit-hash
      </a>
    </p>
    <p
      class="small mb-0"
    >
      <span>
        Logo by 
      </span>
      <a
        href="http://thenounproject.com/term/ticket/92194/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Ramón G.
      </a>
      <span>
         under CC-BY 3.0
      </span>
      <br />
      <span>
        Other icons from 
      </span>
      <a
        href="https://octicons.github.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub Octicons
      </a>
    </p>
  </div>
</DocumentFragment>
`;
